Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -157,68 +234,107 @@ private function _processRequest(array $aRequestParameters)
     /**
      * Returns an array of processing actions to be displayed below the amount input field.
      *
+     * @throws Exception in case the payment method type is not found
+     *
      * @return array
      *
-     * @since 1.0.0
+     * @since 1.0.1
      */
     protected function _getPostProcessingActions(): array
     {
-        // TODO return only supported actions for the payment
-        return [
-            'cancel' => [
-                'title' => Helper::translate('cancel'),
-                'callback' => '_onCancel',
-            ],
-            'capture' => [
-                'title' => Helper::translate('pay'),
-                'callback' => '_onCapture',
-            ],
-            'refund' => [
-                'title' => Helper::translate('refund'),
-                'callback' => '_onRefund',
-            ],
-        ];
+        // array containing all possible post-processing operations on the currently selected transaction
+        $aOperations = [];
+
+        $sPaymentId = $this->oTransaction->getPaymentType();
+        $oPayment = PaymentMethodHelper::getPaymentById($sPaymentId);
+
+        // it is necessary to create a new empty transaction with the ID of the currently
+        // selected one in order to get the available post-processing operations
+        $oPaymentMethod = PaymentMethodFactory::create($sPaymentId);
+        $oTransaction = $oPaymentMethod->getTransaction();
+        $sParentTransactionId = $this->oTransaction->wdoxidee_ordertransactions__transactionid->value;
+        $oTransaction->setParentTransactionId($sParentTransactionId);
+
+        // load the supported operations array from the backend service
+        $oBackendService = new BackendService($oPaymentMethod->getConfig($oPayment), $this->_oLogger);
+        $aSupportedOperations = $oBackendService->retrieveBackendOperations($oTransaction, true);
```

Comment:
I see no real need of creating a new variable as we could just make use of `$aOperations` (but rename as mentioned in my comment above).

Reply:
👍